include/seqan3/alignment/matrix/detail/matrix_coordinate.hpp

```diff
@@ -87,9 +108,33 @@ struct matrix_index
      * \param row_idx The row index to set.
      * \param col_idx The column index to set.
      */
-    constexpr matrix_index(row_index_type<index_t> const row_idx,
-                           column_index_type<index_t> const col_idx) noexcept
-        : row{row_idx.get()}, col{col_idx.get()}
+    constexpr matrix_index(row_index_type<index_t> const row_idx, column_index_type<index_t> const col_idx) noexcept :
+        row{row_idx.get()},
+        col{col_idx.get()}
+    {}
+
+    /*!\brief Construction from strongly typed row index and column index over a scalar type when the index is a simd
+     *        vector.
+     *
+     * \tparam scalar_index_t The type of the scalar index type; must model seqan3::arithmetic.
+     *
+     * \param row_idx The row index to set.
+     * \param col_idx The column index to set.
+     *
+     * \details
+     *
+     * This constructor initialises the row and col index which is represented as simd vectors. The vectors are
+     * initialised with the scalar types of the given strong types over the scalar value. This constructor is only
+     * available if `index_t` is a simd vector type.
+     */
+    template <seqan3::arithmetic scalar_index_t>
+    constexpr matrix_index(row_index_type<scalar_index_t> const row_idx,
+                           column_index_type<scalar_index_t> const col_idx) noexcept
+    //!\cond
+        requires simd_concept<index_t>
+    //!\endcond
+        : row{simd::fill<index_t>(static_cast<typename simd_traits<index_t>::scalar_type>(row_idx.get()))},
+          col{simd::fill<index_t>(static_cast<typename simd_traits<index_t>::scalar_type>(col_idx.get()))}
```

Comment:
```suggestion
        : row{simd::fill<index_t>(typename simd_traits<index_t>::scalar_type{row_idx.get()})},
          col{simd::fill<index_t>(typename simd_traits<index_t>::scalar_type{col_idx.get()})}
```

Reply:
No this won't work if the scalar type of the simd type is smaller than the type stored in row_index or column_index

Reply:
I see. You are right. This is a bummer. can you add a comment 

```suggestion
        : row{simd::fill<index_t>(static_cast<typename simd_traits<index_t>::scalar_type>(row_idx.get()))}, // row idx is size_t, but index_t might be of smaller bit size
          col{simd::fill<index_t>(static_cast<typename simd_traits<index_t>::scalar_type>(col_idx.get()))}
```

Reply:
I added the comment above and also added an explicit concept check to require that the scalar_index_t is convertible to the scalar type in the first place.